plugins/autosuspend.koplugin/main.lua

```diff
@@ -142,8 +143,12 @@ function AutoSuspend:onCloseWidget()
     self.task = nil
 
     self:_unschedule_standby()
-    -- allowStandby is necessary, as we do a preventStandby on plugin start
-    UIManager:allowStandby()
+    if not Device:canStandby() then return end
+    -- allowStandby may be necessary, as we do a preventStandby on plugin start
+    while self.prevent_standby_count > 0 do
+        UIManager:allowStandby()
```

Comment:
Shouldn't we use `self:allowStandby()` here? The call here is the counterpart of the initial `self:preventStandby()` called from `_reschedule_standby` through `init`. It seems to me (untested) that using `UIManager:allowStandby` instead of the correct `self:allowStandby` causes the bug.

Then we wouldn't need the `self.prevent_standby_count` here at all.

Reply:
OK, done in [ab90e70](https://github.com/koreader/koreader/pull/8970/commits/ab90e701548d592c1144db0546d2245eb7d3f6c5) :)